lib/elixir/lib/macro.ex

```diff
@@ -585,6 +585,9 @@ defmodule Macro do
   end
 
   # Access
+  def to_string({{:., _, [Access, :get]}, _, [{_, _, [_|_]} = left, right]} = ast, fun) do
```

Comment:
I think you could limit this function "head" to only match on `@binary_ops` and this should help to keep `Macro.to_string(quote do: Dict.put(%{}, "foo", "bar")["foo"])` intact.